### PR TITLE
Feat/pda 17/code formatting tooling

### PR DESCRIPTION
**Pull request by knzeng-e**
### Summary

|             |   |
|-------------|---|
| Description | This PR adds some code formatting tools to enforce code readability and increase homogeneous format.|
| Jira issue  | https://energyweb.atlassian.net/browse/PDA-17  |

#### Type of request
- [x] New feature (non-breaking change which adds functionality)

### List of Features

- Prettier config is added (find [here](https://prettier.io/docs/en/options.html) the different options that can be set for prettier)
- Eslint config is updated to integrate prettier checks
- Prettier and eslint are run as a precommit-hook using husky

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "scripts": {
     "start-rpc": "run-with-testrpc -m \"candy maple cake sugar pudding cream honey rich smooth crumble sweet treat\" --port 8544 --chainId 1337 --accounts 20 --gasLimit=10000000",
     "ganache": "ganache-cli -m \"candy maple cake sugar pudding cream honey rich smooth crumble sweet treat\" --port 8544 --accounts 20 --networkId=9 --gasLimit=10000000",
-    "prepare": "npm run build",
+    "prepare": "npm run build && husky install",
```
Comment (jrhender) [inline package.json:13]: Why `husky install` on `prepare`? It seems `prepare` is run before `npm install` and `npm publish`?

**Review by JGiter — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (knzeng-e) [reply package.json:13]: I saw in [documentation](https://typicode.github.io/husky/#/) and on the [github page](https://github.com/typicode/husky#usage) that setting it on the `prepare script` enables git hooks directly after the install and long before `npm publish`

Comment (knzeng-e) [reply package.json:13]: The consequence of that is that at each commit, the eslint & prettier checks will be fired, and the commit wont be possible if there is any error.

Comment (JGiter) [reply package.json:13]: It will create husky.sh in .husky/_/

Comment (jrhender) [reply package.json:13]: Thanks!

**Review by jrhender — APPROVED** [review]

**Pull request merged**